R/build-favicons.R

```diff
@@ -23,15 +13,33 @@ copy_favicons <- function(pkg = ".") {
 #' during package checking.
 #'
 #' @inheritParams as_pkgdown
+#' @param clobber If `TRUE`, re-create favicons from package logo.
 #' @export
-build_favicon <- function(pkg = ".") {
+build_favicons <- function(pkg = ".", clobber = FALSE) {
   pkg <- as_pkgdown(pkg)
 
+  rule("Building favicons")
+
   logo_path <- find_logo(pkg$src_path)
+
   if (is.null(logo_path)) {
-    stop("Can't find package logo.", call. = FALSE)
+    message("Can't find package logo PNG or SVG to build favicons.")
+    return()
   }
 
+  if (has_favicons(pkg) && !clobber) {
+    message("Favicons already exist in `pkgdown/`. Copying to site.")
```

Comment:
This doesn't actually copy anything, so I think the message needs to go somewhere else.